hack/generated/pkg/testcommon/error_translating_roundtripper.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package testcommon
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/dnaeon/go-vcr/recorder"
+	"github.com/google/go-cmp/cmp"
+)
+
+// translateErrors wraps the given Recorder to handle any "Requested interaction not found"
+// and log better information about what the expected request was.
+//
+// By default the error will be returned to the controller which might ignore/retry it
+// and not log any useful information. So instead here we find  the recorded request with
+// the body that most closely matches what was sent and report the "expected" body.
+//
+// Ideally we would panic on this error but we don't have a good way to deal with the following
+// problem at the moment:
+// - during record the controller does GET (404), PUT, … GET (OK)
+// - during playback the controller does GET (which now returns OK), DELETE, PUT, …
```

Comment:
I thought that the ordering/count header existed specifically to make the first GET here 404?

Reply:
Right, let me get that restored and see if I can make this all work properly...

Reply:
Since you fixed the counting header, is this still true or does this comment need updating?

Reply:
There’s still one rogue DELETE happening and I'm not sure why. Will stick an issue to investigate later...

Reply:
https://github.com/Azure/azure-service-operator/issues/1497